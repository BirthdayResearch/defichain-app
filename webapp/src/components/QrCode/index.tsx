import React, { useState } from 'react';
import { Button, Popover, PopoverBody } from 'reactstrap';
import QRCode from 'qrcode.react';
import { AiOutlineQrcode } from 'react-icons/ai';

interface QrCodeProps {
  value: string | undefined;
  uid: string;
  qrClass: string;
}

const QrCode: React.FunctionComponent<QrCodeProps> = (props: QrCodeProps) => {
  const [qrOpen, toggleQR] = useState(false);

  return (
    <>
      <Button
        color='link'
        size='sm'
        className='padless ml-2'
        id={props.uid}
        onClick={e => e.currentTarget.focus()} // Need for Popover trigger="focus" to work
      >
        <AiOutlineQrcode />
      </Button>
      <Popover
        placement='auto'
        isOpen={qrOpen}
        target={props.uid}
        toggle={() => toggleQR(!qrOpen)}
        trigger='focus'
      >
        <PopoverBody>
          <QRCode value={props.value} size={240} className={props.qrClass} />
        </PopoverBody>
      </Popover>
    </>
  );
};

export default QrCode;
