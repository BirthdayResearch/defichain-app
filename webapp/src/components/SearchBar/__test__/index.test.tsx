import React from 'react';
import QrCode from '../index';
import { shallow } from 'enzyme';

describe('QrCode component', () => {
  it('should check for snapshot', () => {
    const props = {
      searching: 'searching',
      toggleSearch: 'toggleSearch',
      onChange: () => {},
      placeholder: 'placeholder',
    };
    const wrapper = shallow(<QrCode {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
