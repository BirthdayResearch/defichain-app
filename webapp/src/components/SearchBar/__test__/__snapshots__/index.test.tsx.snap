// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QrCode component should check for snapshot 1`] = `
<div
  className="d-block"
>
  <FormGroup
    className="row undefined"
    tag="div"
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <InputGroup
        tag="div"
      >
        <Input
          id="searchInput"
          name="searchInput"
          onChange={[Function]}
          placeholder="placeholder"
          type="text"
        />
        <MdSearch />
        <InputGroupAddon
          addonType="append"
          tag="div"
        >
          <Button
            color="outline-primary"
            onClick="toggleSearch"
            tag="button"
          >
            <MdClose />
          </Button>
        </InputGroupAddon>
      </InputGroup>
    </Col>
  </FormGroup>
</div>
`;
